grafana-plugin/src/state/rootBaseStore/index.ts

```diff
@@ -211,25 +207,17 @@ export class RootBaseStore {
         }
       }
     } else {
-      const syncDataResponse = await PluginState.syncDataWithOnCall(this.onCallApiUrl);
```

Comment:
we can remove everything associated with `PluginState.syncDataWithOnCall`, it's only used here. As well as `PluginState.pollOnCallDataSyncStatus` and `PluginState._pollOnCallDataSyncStatus`.

I _believe_ this also means we can mark the `POST /api/internal/v1/plugin/sync` endpoint as deprecated as well. Which means the whole `PluginSyncView` is deprecated 🥳  (double check me on this, but I think this is the case)